objax/io/checkpoint.py

```diff
@@ -107,6 +116,8 @@ def save(self, vc: VarCollection, idx: int):
             vc: variables collection to save.
             idx: index of the new checkpoint where variables should be saved.
         """
+        if not self.do_writes:
```

Comment:
On second thought, I'm wondering if modifying `checkpoint.py` is the right thing to do. It adds complexity and it's not intuitive. In fact, it seems better suited to only save in imagenet training loop if `host_id == 0`.